### PR TITLE
Added: http request context handling

### PR DESCRIPTION
**Pull request by lwoydziak**
see: https://github.com/pivotal-cf/brokerapi/issues/29
[#132019309](https://www.pivotaltracker.com/story/show/132019309)

### CODE REVIEW

**Comment by lwoydziak** [comment]
Alternative impl:
https://github.com/pivotal-cf/brokerapi/pull/31

**Comment by ablease** [comment]
Thanks for this! We prefer this implementation to #31 as it follows the pattern defined in the[ standard library](https://golang.org/pkg/context/), specifically:

`Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:`

After discussing this with our PM, we are happy to have a breaking change to our `ServiceBroker` interface to support Contexts, and support go 1.7 as a minimum going forward.  Therefore we do not need the backwards compatibility you've added to this PR.

In summary:
- Change context.Context to the first parameter on the ServiceBroker interface functions
- Remove the support for backwards compatibility, and explicitly support context on the ServiceBroker interface.

Cheers,

Alex & @Samze

**Comment by lwoydziak** [comment]
Great, I'll make the changes today.

**Comment by Samze** [comment]
Thank you @lwoydziak for the making the changes. We think we should add the HTTP request context to `LastOperation` as well, unless you can think of a reason not to?

**Comment by lwoydziak** [comment]
@Samze Good point, after further conversation around here, it makes sense to have them on all calls as we are not sure what the implementation will end up doing. - Uploaded a patch to reflect that.

**Comment by craigfurman** [comment]
Thank you @lwoydziak
